core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -1018,15 +1018,15 @@ else if (physicalOffset < 0) {
             return builder.toString();
         }
 
-        private String formatRowsPerMatch(RowsPerMatch rowsPerMatch)
```

Comment:
I liked the previous more as it was more generic

Reply:
How is it more generic? It seems perfectly equivalent to me...

Reply:
The alternative is to use the `nodeOutput.appendDetails("%s", formatRowsPerMatch(...))` pattern :)

Reply:
It was more generic as you could anything with the returned value. Now the value is always consumed the same way.

Reply:
OK, I'll drop the two commits which refactor these methods and I'll "paste" the return value into a simple format string instead.